### ISSUE TITLE
 Add dynamic Primary Hero Shelf to desktop homepage

### ISSUE THREAD

**Issue by jvillalobos** [mozilla/addons-frontend#8305]
Depends on #8478 

Part of [this PRD](https://docs.google.com/document/d/12aD_v5AOj1Y6VmT4xOuyNcrKfZxqhP8R_AOHroQVLV8/edit#). Depends on https://github.com/mozilla/addons-server/issues/11830.

The Primary Hero Shelf in the homepage will feature one extension, chosen from a subset of Recommended extensions:

<img width="623" alt="Screen Shot 2019-07-03 at 4 02 16 PM" src="https://user-images.githubusercontent.com/62568/60627685-16904480-9dac-11e9-8d5b-438676aeeb44.png">

Once the admin for this shelf is done (https://github.com/mozilla/addons-server/issues/11830), we should show it on the homepage.

Every time the homepage is loaded, one of the extensions included in the Primary Hero Shelf set should be featured in this shelf. The image and color for the background gradient are chosen in the admin. The decoration on the bottom left will be chosen at random from a pool of predefined decorations (pending from @brassy-, covered in https://github.com/mozilla/addons-frontend/issues/8248 ).

The layout should look like the mockup above, with a couple of changes:
* It should say `Recommended` instead of `Recommendation`
* Below the add-on name, it should say `by $DEVELOPERS`

The description should be the custom description from the Recommendations admin, not the description from the add-on metadata.

The Call to Action should link to the listing page for listed add-ons, and the homepage URL for unlisted add-ons.

All internal URLs in this module should have these params appended: `src=homepage-primary-hero`. External URLs should have these: `utm_source=addons.mozilla.org&utm_medium=referral&utm_content=homepage-primary-hero`. An event should be logged on GA when the CTA is clicked, recording a click on the shelf and the destination.

**Comment by AlexandraMoga** [comment]
I've verified the primary hero layout and contents on -dev. Below is a demo of the featured add-ons loading dynamically:

![2019-09-02_13-27-34](https://user-images.githubusercontent.com/31961530/64108453-7e9be080-cd85-11e9-9da7-58c57378a1b0.gif)


There were also some issues found (referenced above). 

@bobsilverberg apart from the items mentioned in https://github.com/mozilla/addons-frontend/pull/8487#issue-308185419, is the following aslo not implemented?

>All internal URLs in this module should have these params appended: src=homepage-primary-hero. External URLs should have these: utm_source=addons.mozilla.org&utm_medium=referral&utm_content=homepage-primary-hero. An event should be logged on GA when the CTA is clicked, recording a click on the shelf and the destination.

The only thing that I've noticed was that the call to action button for listed add-ons is mapped to `https://addons-dev.allizom.org/en-US/firefox/addon/addon-slug/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=homepage-primary-hero`, but this is not fulfilling the requirement for internal URLs mentioned above.

**Comment by bobsilverberg** [comment]
> 
> @bobsilverberg apart from the items mentioned in [#8487 (comment)](https://github.com/mozilla/addons-frontend/pull/8487#issue-308185419), is the following aslo not implemented?
> 
> > All internal URLs in this module should have these params appended: src=homepage-primary-hero. External URLs should have these: utm_source=addons.mozilla.org&utm_medium=referral&utm_content=homepage-primary-hero. An event should be logged on GA when the CTA is clicked, recording a click on the shelf and the destination.
> 
> The only thing that I've noticed was that the call to action button for listed add-ons is mapped to `https://addons-dev.allizom.org/en-US/firefox/addon/addon-slug/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=homepage-primary-hero`, but this is not fulfilling the requirement for internal URLs mentioned above.

Thanks @AlexandraMoga, I think I missed that requirement when I was converting the static page into a dynamic page. Could you please open a follow-up issue for that?

**Comment by AlexandraMoga** [comment]
@bobsilverberg I filed #8537 for the additional requirement.

**Comment by AlexandraMoga** [comment]
I'm marking this issue as verified fixed because we have follow ups for all the outstanding issues found so far.
